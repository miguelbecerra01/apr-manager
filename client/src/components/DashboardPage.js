import React from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { getApiHost } from '../utils';

import * as moment from 'moment/moment';
moment.locale('es');

export class DashboardPage extends React.Component {
    state = {
        statements: []
    }
    componentDidMount() {
        axios.get(`${getApiHost()}/api/statements`)
            .then(res => {
                const statements = res.data;
                this.setState({ statements });
            });
    }
    render() {
        return (
            <div>
                Dashboard page content your private contents  heress
                <Link to="/statements"> Pagar mi cuenta como invitado</Link>
                {this.state.statements.map((statement, index) => {
                    return (
                        <p key={index}>
                            {statement.id_member}
                            {statement.total_amount}
                            {statement.status}
                            {statement.emission_date}
                            {statement.due_date}
                        </p>
                    )
                })}
            </div>
        );
    }
}


export default DashboardPage;