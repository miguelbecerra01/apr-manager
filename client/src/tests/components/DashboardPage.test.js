import React from 'react';
import { shallow } from 'enzyme';
import DashboardPage from '../../components/DashboardPage';

test('should render DashboardPage correctly', () => {
    global.window = jest.fn(() => ({
        location: {
            prototol: 'http://'
        }
    }));

    const wrapper = shallow(<DashboardPage />);
    expect(wrapper).toMatchSnapshot();
});