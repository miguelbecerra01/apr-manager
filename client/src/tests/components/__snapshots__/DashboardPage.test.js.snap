// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<div>
  Dashboard page content your private contents  heress
  <Link
    replace={false}
    to="/statements"
  >
     Pagar mi cuenta como invitado
  </Link>
</div>
`;
